### PR TITLE
Cover e2e scenarios in enzyme tests

### PR DESCRIPTION
**Pull request by sjanota**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: service-catalog-ui/catalog/src/components/ServiceClassDetails/CreateInstanceModal/test/CreateInstanceModal.test.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+          formElementRef={{ current: null }}
+          jsonSchemaFormRef={{ current: null }}
+          onChange={onChange}
+          onError={onError}
+          onCompleted={onCompleted}
+        />
+      </MockedProvider>,
+    );
+
+    await componentUpdate(component);
```
[the remark this answers is not in the document]
Comment (akucharska) [reply service-catalog-ui/catalog/src/components/ServiceClassDetails/CreateInstanceModal/test/CreateInstanceModal.test.js]: Some how it doesn't work without two updates. I guess it only forces one state update and not all of them

File: service-catalog-ui/catalog/src/components/ServiceClassList/searchUtils.js [hunk 2]
```diff
@@ -20,23 +21,10 @@ const determineDisplayedServiceClasses = (
...
-  if (tabIndex === serviceClassConstants.servicesIndex) {
-    filteredByTab = filteredByLabels.filter(item => {
-      if (item.labels) {
-        return item.labels.local !== 'true';
-      }
-      return true;
-    });
-  }
+  const filterFunction =
+    tabIndex === serviceClassConstants.addonsIndex ? isAddon : isService;
```
Comment (dariadomagala) [inline service-catalog-ui/catalog/src/components/ServiceClassList/searchUtils.js]: Shouldn't it be the other way around?

File: service-catalog-ui/instances/src/commons/helpers.js [hunk 3]
```diff
@@ -140,3 +140,19 @@ export class DocsProcessor {
...
   }
 }
+
+function getServiceClass(instance) {
+  return instance.serviceClass
+    ? instance.serviceClass
+    : instance.clusterServiceClass;
+}
+
+export function isAddon(instance) {
```
Comment (dariadomagala) [inline service-catalog-ui/instances/src/commons/helpers.js]: We don't need `isAddon` since we only use `isService` function.

Comment (akucharska) [reply service-catalog-ui/instances/src/commons/helpers.js]: We use isAddon in searchUtils

File: service-catalog-ui/catalog/src/components/ServiceClassList/searchUtils.js (hunk 2 above)
Comment (akucharska) [reply service-catalog-ui/catalog/src/components/ServiceClassList/searchUtils.js]: No. On addon tab we filter for addons

Comment (dariadomagala) [reply service-catalog-ui/catalog/src/components/ServiceClassList/searchUtils.js]: I meant more like `isAddon ? serviceClassConstants.addonsIndex : ...`

Comment (akucharska) [reply service-catalog-ui/catalog/src/components/ServiceClassList/searchUtils.js]: No. This is filtering function. Based on chosen tab (tabIndex === serviceClassConstants.addonsIndex) we use function that return true for addons, or true for services

**Pull request merged**